airgun/browser.py

```diff
@@ -409,3 +410,45 @@ def save_downloaded_file(self, file_uri=None, save_path=None):
         self.url = current_url
         self.plugin.ensure_page_safe()
         return file_path
+
+    def get_alert(self):
+        """Returns the current alert/PF4 alert object.
+
+        Raises:
+            :py:class:`selenium.common.exceptions.NoAlertPresentException`
+        """
+        locator = "//div[@data-ouia-component-type='PF4/ModalContent']"
+        if not self.handles_alerts:
+            return None
+        try:
+            if self.move_to_element(locator).is_displayed():
+                return Pf4ConfirmationDialog(self.browser)
+        except NoSuchElementException:
+            self.logger.info("No PF4 alert detected !")
+        return self.selenium.switch_to.alert
+
+    def handle_alert(
+        self,
+        cancel=False,
+        wait=30.0,
+        squash=False,
+        prompt=None,
+        check_present=False,
+    ):
+        """Extend the behaviour of widgetstatic.browser.handle_alert to handle PF4 alerts"""
+        popup = self.get_alert()
+        if isinstance(popup, Pf4ConfirmationDialog):
+            if cancel:
+                self.logger.info("  dismissing")
+                popup.cancel()
+            else:
+                self.logger.info("  accepting")
+                popup.confirm()
+        else:
+            self.handle_alert(
```

Comment:
Are you intending to call an ancestor's `handle_alert`, or make this recursive? I think its the first, in which case I'd call it with `super()` instead

Reply:
I guess I miss the trick here need to use the `super()`

Reply:
https://docs.python.org/3/library/functions.html#super

So `super()` will resolve the parent class instance, and you can be sure which class's `handle_alert` method is called.